src/Microsoft.Data/StringColumn.BinaryOperations.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public partial class StringColumn : BaseColumn
+    {
+        public override BaseColumn Add(BaseColumn column)
+        {
+            // TODO: Using indexing is VERY inefficient here. Each indexer call will find the "right" buffer and then return the value
+            if (Length != column.Length)
+            {
+                throw new ArgumentException($"Column lengths are mismatched", nameof(column));
+            }
+            StringColumn ret = Clone();
+            for (long i = 0; i < Length; i++)
+            {
+                ret[i] += column[i].ToString();
+            }
+            return ret;
+        }
+
+        public override BaseColumn Add<T>(T value)
+        {
+            StringColumn ret = Clone();
+            string valString = value.ToString();
+            for (int i = 0; i < ret.stringBuffers.Count; i++)
+            {
+                IList<string> buffer = ret.stringBuffers[i];
+                int bufferLen = buffer.Count;
+                for (int j = 0; j < bufferLen; j++)
+                {
+                    buffer[j] += valString;
+                }
+            }
+            return ret;
+        }
+
+        public override BaseColumn Equals(BaseColumn column)
+        {
+            // TODO: Using indexing is VERY inefficient here. Each indexer call will find the "right" buffer and then return the value
+            if (Length != column.Length)
+            {
+                throw new ArgumentException($"Column lengths are mismatched", nameof(column));
```

Comment:
Should we be adding the exception messages (and any other localizable strings) to a .resx file from the start? It will be easier to add them now, then go through all the strings and try to do it after the fact.

Reply:
Oh good idea. I'll do that later this week, once I'm done with app building